examples/vanilla-js/app.js

```diff
@@ -0,0 +1,80 @@
+import { ContentfulLivePreview } from '@contentful/live-preview';
+import { createClient } from 'contentful';
+import dotenv from 'dotenv';
+
+// Configuration object
+const CONFIG = {
+  locale: 'en-US',
+  entryId: 'W9z8t0uXz3CMFjp4sw5zj',
+  fields: ['title'],
+  subscriptions: [],
+  debugMode: true,
+};
+
+document.addEventListener('DOMContentLoaded', initialize);
+
+function initialize() {
+  dotenv.config();
+
+  const client = createClient({
+    space: process.env.CONTENTFUL_SPACE_ID,
+    accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
+    host: 'preview.contentful.com',
+  });
+
+  ContentfulLivePreview.init({ locale: CONFIG.locale });
+
+  client
+    .getEntry(CONFIG.entryId)
+    .then((entry) => {
+      CONFIG.fields.forEach((fieldId) => {
+        displayFieldData(entry, client, fieldId);
+        setupLivePreview(entry, fieldId);
+      });
+    })
+    .catch((err) => console.error(err));
+}
+
+function setAttributes(element, attributes) {
+  for (const [key, value] of Object.entries(attributes)) {
+    // Skip undefined or null values
+    if (value === undefined || value === null) continue;
+
+    if (key.startsWith('data-')) {
+      const formattedName = key.split('data-')[1].replace(/-([a-z])/g, (m, w) => w.toUpperCase());
+      element.dataset[formattedName] = value;
+    } else {
+      element.setAttribute(key, value);
+    }
+  }
+}
+
+function displayFieldData(entry, client, fieldId) {
+  const domElement = document.getElementById(fieldId);
+  if (!domElement) {
+    console.error(`DOM element with ID "${fieldId}" not found.`);
+    return;
+  }
+
+  const props = ContentfulLivePreview.getProps({ entryId: CONFIG.entryId, fieldId });
+  setAttributes(domElement, props);
+
+  domElement.textContent = entry.fields[fieldId];
+}
+
+function setupLivePreview(entry, fieldId) {
+  const callback = (updatedData) => {
+    const domElement = document.getElementById(fieldId);
+    if (domElement && updatedData.fields && updatedData.fields[fieldId]) {
+      domElement.textContent = updatedData.fields[fieldId];
```

Comment:
should we check if it's text? and otherwise call a function to wait 5s and reload?
Guess that could be an use-case that would work for some

Reply:
ah see, our example is only for one text 🤔

Reply:
Yes all our examples are just for text fields. To keep things simple.
I can add a check if it's text though